src/wallet/db.cpp

```diff
@@ -692,8 +692,10 @@ void CDBEnv::Flush(bool fShutdown)
             if (mapFileUseCount.empty()) {
                 dbenv->log_archive(&listp, DB_ARCH_REMOVE);
                 Close();
-                if (!fMockDb)
+                if (!fMockDb) {
                     fs::remove_all(fs::path(strPath) / "database");
+                }
+                g_dbenvs.erase(strPath);
```

Comment:
Might want to consider picking 0b82bac76d0f842bd2294a290388536951fbc576, which also removes a dangling `env` instance in `CWalletDBWrapper::Flush`

Reply:
yeah, looks good 👍 .

Reply:
will backport only the dangling env instance fix here and leave the load/unload functionality for a follow-up PR. This one is quite large already.